astropy/modeling/fitting.py

```diff
@@ -499,7 +543,29 @@ def __call__(self, model, x, y, z=None, weights=None, rcond=None):
 
         masked = np.any(np.ma.getmask(rhs))
 
-        if len(model_copy) == 1 or not masked:
+        if weights is not None and weights.ndim > 1 and weights.shape == rhs.shape:
```

Comment:
I think you don't need the 3rd condition here (`shape`), because `weights` can only have a model set axis at this point if the `rhs` also has one (and their shapes should have been checked earlier).

Reply:
Ok, I removed the condition.